ng2-components/ng2-activiti-processlist/src/components/activiti-start-process.component.spec.ts

```diff
@@ -80,79 +72,217 @@ describe('ActivitiStartProcessButton', () => {
         window['componentHandler'] = componentHandler;
     });
 
-    it('should display the correct number of processes in the select list', () => {
-        fixture.detectChanges();
-        let selectElement = debugElement.query(By.css('select'));
-        expect(selectElement.children.length).toBe(3);
-    });
+    describe('process definitions list', () => {
 
-    it('should display the correct process def details', (done) => {
-        fixture.detectChanges();
-        fixture.whenStable().then(() => {
-            let optionEl: HTMLOptionElement = debugElement.queryAll(By.css('select option'))[1].nativeElement;
-            expect(optionEl.value).toBe('my:process1');
-            expect(optionEl.textContent.trim()).toBe('My Process 1');
-            done();
+        it('should call service to fetch process definitions', () => {
+            fixture.detectChanges();
+            expect(getDefinitionsSpy).toHaveBeenCalled();
```

Comment:
I have left this test in after verifying that the HTML is also checked (check correct number/content of `option` elements in later tests. There is no need to add an additional output to detect when the results have been returned for the purposes of the test since `whenStable()` ensures that the test waits until all async activity has completed before making assertions on the HTML.